lib/mix/tasks/nettools.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Mix.Tasks.Archethic.Nettools do
+  @moduledoc """
+  Provides Support for Networking tools At cli
+
+  ## Command line options
+
+    * `-h`, `--help`  - show this help
+    * `-p`, `--punch` - Attempt to open ports
+    * `-i`, `--ip`    - Public IP
+
+  ## Command line arguments
+    * `punch` - args: []-> 30002 40002, list_of ports
+
+  ## Example
+
+  ```sh
+  mix archethic.nettools -p
+  mix archethic.nettools --punch
+  mix archethic.nettools -p 30_000 40_000
+  mix archethic.nettools -i
+  mix archethic.nettools --ip
+  ```
+
+  """
+  use Mix.Task
+  alias Archethic.Networking
+
+  require Logger
+
+  @impl Mix.Task
+  def run(args) do
+    args
+    |> parse_args
+  end
+
+  def parse_args(args) do
+    OptionParser.parse(args,
+      strict: [help: :boolean, punch: :boolean, ip: :boolean],
+      aliases: [h: :help, p: :punch, i: :ip]
+    )
+    |> args_to_internal_representation()
+  end
+
+  def args_to_internal_representation({[help: true], [], []}) do
+    Mix.shell().cmd("mix help #{Mix.Task.task_name(__MODULE__)}")
+  end
+
+  def args_to_internal_representation({[punch: true], _port_list = [port | more_ports], errors}) do
+    Networking.PortForwarding.try_open_port(abs(Integer.parse(port) |> elem(0)), true)
+    args_to_internal_representation({[punch: true], more_ports, errors})
```

Comment:
You might also need to set env variable as well otherwise this will be skipped.

Also `PortForwarding` must be updated at https://github.com/archethic-foundation/archethic-node/blob/165fc8c2664037b25d0d6e7d0b3b91055d2b439d/lib/archethic/networking/port_forwarding.ex#L35 to set `NATDiscovery`

Reply:
Sincere apologies. my bad,

Reply:
No harm 😉